daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/TextNumberParsers.scala

```diff
@@ -0,0 +1,162 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.processors.parsers
```

Comment:
This file isn't actually entirely new. I just rename TextNumberPrimitives1.scala to a more appropriate TextNumberParsers. I guess GitHub can't detect that rename. The bulk of this is the same except for using the new evaluatable, using primNumeric for range checking/type conversion, and tweaks to error messages to make them more consistent.

Reply:
I have made this mistake repeatedly as well - renaming things, then finding that the PR can't diff things properly. 

I now advocate renaming files as a last separate commit, or even a separate PR. It's hard to resist when you are cleaning things up though. Particularly if you expect git to detect the rename, which it sometimes does.